app/models/service_template/copy.rb

```diff
@@ -6,12 +6,24 @@ def template_copy(new_name = "Copy of " + name + Time.zone.now.to_s)
       ActiveRecord::Base.transaction do
         dup.tap do |template|
           template.update_attributes(:name => new_name, :display => false)
-          service_resources.each do |sr|
-            resource = sr.resource.respond_to?(:service_template_resource_copy) ? sr.resource.service_template_resource_copy : sr.resource
-            template.add_resource(resource, sr)
-          end
+          service_resources.each { |sr| resource_copy(sr, template) }
+          custom_buttons.each { |cb| custom_button_copy(cb, template) }
+          custom_button_sets.each { |cbs| custom_button_set_copy(cbs, template) }
         end.save!
       end
     end
   end
+
+  def resource_copy(sr, template)
+    resource = sr.resource.respond_to?(:service_template_resource_copy) ? sr.resource.service_template_resource_copy : sr.resource
+    template.add_resource(resource, sr)
+  end
+
+  def custom_button_copy(cb, template)
+    cb.copy(:name => "Copy of " + cb.name + Time.zone.now.to_s, :description => "Copy of " + cb.description + " for template #{template.id}")
```

Comment:
Buttons will be applied to a new service_template, the names should display the same as the original, there is no need to focus a unique name/description.

Also, you are copying the buttons but they are not being assigned to the new template.